src/cdk/table/table.ts

```diff
@@ -459,24 +460,28 @@ export class CdkTable<T> implements AfterContentChecked, CollectionViewer, OnDes
     }
 
     // Render updates if the list of columns have been changed for the header, row, or footer defs.
-    this._renderUpdatedColumns();
+    let forced = this._renderUpdatedColumns();
 
     // If the header row definition has been changed, trigger a render to the header row.
     if (this._headerRowDefChanged) {
       this._forceRenderHeaderRows();
       this._headerRowDefChanged = false;
+      forced = true;
     }
 
     // If the footer row definition has been changed, trigger a render to the footer row.
     if (this._footerRowDefChanged) {
       this._forceRenderFooterRows();
       this._footerRowDefChanged = false;
+      forced = true;
     }
 
     // If there is a data source and row definitions, connect to the data source unless a
     // connection has already been made.
     if (this.dataSource && this._rowDefs.length > 0 && !this._renderChangeSubscription) {
       this._observeRenderChanges();
+    } else if (forced) {
```

Comment:
I can add a comment - the else if is because if the table has not rendered any rows yet (the first condition, basically), then this will be taken care of the first time that the table receives row data.

Without this, we'd compute sticky columns and then have to throw that out and do it again when that data arrives.

Reply:
Comment added